jdaviz/configs/default/plugins/export/export.py

```diff
@@ -258,6 +259,46 @@ def _is_filename_changed(self, event):
         # Clear overwrite warning when user changes filename
         self.overwrite_warn = False
 
+    def _update_subset_format_disabled(self):
+        new_items = []
+        if self.subset.selected is not None:
+            subset = self.app.get_subsets(self.subset.selected)
+            if self.app._is_subset_spectral(subset[0]):
+                good_formats = ["ecsv"]
+            else:
+                good_formats = ["fits", "reg"]
+            for item in self.subset_format_items:
+                if item["label"] in good_formats:
+                    item["disabled"] = False
+                else:
+                    item["disabled"] = True
+                    if item["label"] == self.subset_format.selected:
+                        self.subset_format.selected = good_formats[0]
+                new_items.append(item)
+        self.subset_format_items = []
+        self.subset_format_items = new_items
+
+    @observe('subset_format_selected')
+    def _disable_subset_format_combo(self, event):
+        # Disable selecting a bad subset+format combination from the API
+        if self.subset.selected == '' or self.subset.selected is None:
+            return
+        subset = self.app.get_subsets(self.subset.selected)
+        bad_combo = False
+        if self.app._is_subset_spectral(subset[0]):
+            if event['new'] != "ecsv":
+                bad_combo = True
+        elif event['new'] == "ecsv":
+            bad_combo = True
+
+        if bad_combo:
+            # Set back to a good value and raise error
+            good_format = [format["label"] for format in self.subset_format_items if
+                           format["disabled"] is False][0]
+            self.subset_format.selected = good_format
+            raise ValueError(f"Cannot export {self.subset.selected} in {event['new']}"
+                             f" format, reverting selection to {self.subset_format.selected}")

```

Comment:
@kecnry, do you think this error message makes clear enough what is happening?

Reply:
if you want to go the revert route, you could also use `self.subset_format.select_previous()` (which hopefully should be valid still... I can't think of a case where it wouldn't)

Reply:
but the message makes sense to me, let me test it to see if I can find a way to break it and otherwise should be good to approve.